### DISCUSSION TITLE
With Mode.ORBIT how to set 3d model only rotation horizontal

### DISCUSSION THREAD

**Discussion by haothanhit** [google/filament#7472] [Q&A]
i want to prevent the user from being able to rotate my character vertically. I want them to only be able to rotate horizontally

**Comment by haothanhit** [comment]
help me in android code

**Reply by thanhhaoPG** [reply]
@romainguy help you bro!

**Comment by poweifeng** [comment]
You can take a look at the [OrbitManipulator] to see how that's implemented and modify that to suit your needs.  

[OrbitManipulator]: https://github.com/google/filament/blob/main/libs/camutils/src/OrbitManipulator.h
